### ISSUE TITLE
Postgres: ALTER TYPE throws a "Found unparsable section" error.

### ISSUE THREAD

**Issue by MaximeCheramy** [sqlfluff/sqlfluff#3216]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This code:
```sql
ALTER TYPE ats RENAME TO ats__;
```
Generates this error:

```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'ALTER
                       | TYPE ats RENAME TO ats__'
```

### Expected Behaviour

It's valid PostgreSQL syntax so it should work: https://www.postgresql.org/docs/current/sql-altertype.html

### Observed Behaviour

See what happened section.

### How to reproduce

example.sql:
```
ALTER TYPE ats RENAME TO ats__;

```

```
sqlfluff lint --dialect postgres example.sql
```

### Dialect

Postgres

### Version

0.13.0

### Configuration

None.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by jmc-bbk** [comment]
@adityapat3l and @IzieW good first issue.
